editor/editor_file_system.cpp

```diff
@@ -1102,6 +1110,20 @@ void EditorFileSystem::_process_file_system(const ScannedDirectory *p_scan_dir,
 	}
 }
 
+void EditorFileSystem::_process_removed_files_first_scan() {
+	for (const KeyValue<String, EditorFileSystem::FileCache> &kv : file_cache) {
+		if (!processed_files_first_scan.has(kv.key)) {
+			if (ClassDB::is_parent_class(kv.value.type, SNAME("Script"))) {
```

Comment:
If this ever needs more optimization in the future, the results of the `ClassDB` call could be cached locally to avoid repeat queries :)